Formula/cp2k.rb

```diff
@@ -48,35 +57,95 @@ def install
-    inreplace Dir["arch/Darwin-gfortran.*smp"],
+    inreplace Dir["arch/#{arch}-gfortran.*smp"],
               "-lfftw3", "-lfftw3 -lfftw3_threads"
 
+    # Remove flags for unused dependencies on Linux.
+    # The build system has been completely refactored upstream and
+    # these manual fixes will not be needed in the next release.
+    unless OS.mac?
+      inreplace Dir["arch/#{arch}-gfortran.*smp"] do |s|
+        s.gsub!("-D__LIBVORI", "")
+        s.gsub!("-D__LIBXSMM", "")
+        s.gsub!("-D__SPGLIB", "")
+
+        s.gsub!("-I$(LIBXSMM_INC)", "")
+
+        s.gsub!("$(LIBVORI_LIB)/libvori.a", "")
+        s.gsub!("$(SPGLIB_LIB)/libsymspg.a", "")
+        s.gsub!("$(LIBXSMM_LIB)/libxsmmf.a", "")
+        s.gsub!("$(LIBXSMM_LIB)/libxsmm.a", "")
+        s.gsub!("$(LIBPATH)/liblapack.a", "")
+
+        s.gsub!("$(GNU_PATH)/fftw/3.3.9/include", Formula["fftw"].opt_include)
+        s.gsub!("$(GNU_PATH)/fftw/3.3.9/lib", Formula["fftw"].opt_lib)
+        s.gsub!("$(GNU_PATH)/libint/2.6.0-lmax-6", libexec)
+        s.gsub!("$(GNU_PATH)/libxc/5.1.4/include", Formula["libxc"].opt_include)
+        s.gsub!("$(GNU_PATH)/libxc/5.1.4/lib", Formula["libxc"].opt_lib)
+      end
+
+      inreplace "arch/#{arch}-gfortran.ssmp" do |s|
+        s.gsub!("$(LIBPATH)/libblas.a", "#{Formula["openblas"].opt_lib}/libopenblas.a")
+        s.gsub!("-static", "")
+      end
+
+      inreplace "arch/#{arch}-gfortran.psmp" do |s|
+        s.gsub!("include       $(MPI_PATH)/plumed2/2.6.2/lib/plumed/src/lib/Plumed.inc.static", "")
+        s.gsub!("-D__ELPA", "")
+        s.gsub!("-D__PLUMED2", "")
+        s.gsub!("-D__SIRIUS", "")
+        s.gsub!("-D__SCALAPACK", "")
+
+        s.gsub!("-I$(ELPA_INC)/elpa -I$(ELPA_INC)/modules", "")
+        s.gsub!("-I$(SIRIUS_INC)", "")
+        s.gsub!("-I$(SPGLIB_INC)", "")
+
+        s.gsub!("$(PLUMED_DEPENDENCIES)", "")
+        s.gsub!("$(ELPA_LIB)/libelpa_openmp.a", "")
+        s.gsub!("${SIRIUS_LIB}/libsirius.a", "")
+        s.gsub!("$(GNU_PATH)/SpFFT/0.9.13/lib/libspfft.a", "")
+        s.gsub!("$(GNU_PATH)/SpLA/1.2.1/lib/libspla.a", "")
+        s.gsub!("$(GNU_PATH)/hdf5/1.12.0/lib/libhdf5.a", "#{Formula["hdf5"].opt_lib}/libhdf5.a")
+        s.gsub!("$(GNU_PATH)/OpenBLAS/0.3.15/lib/libopenblas.a",
+                "#{Formula["openblas"].opt_lib}/libopenblas.a")
+        s.gsub!("$(LIBPATH)/libz.a", "#{Formula["zlib"].opt_lib}/libz.a")
+        s.gsub!("$(MPI_LIBRARY_PATH)/libscalapack.a", "")
+      end
+    end
+
     # Now we build
     %w[ssmp psmp].each do |exe|
-      # Issue with parallel build: https://github.com/cp2k/cp2k/issues/1316
-      ENV.deparallelize { system "make", "ARCH=Darwin-gfortran", "VERSION=#{exe}" }
-      bin.install "exe/Darwin-gfortran/cp2k.#{exe}"
-      bin.install "exe/Darwin-gfortran/cp2k_shell.#{exe}"
+      system "make", "ARCH=#{arch}-gfortran", "VERSION=#{exe}",
+                     "GSL_LIBRARY_DIR=#{Formula["gsl"].opt_lib}"
```

Comment:
Adding this flag on macOS where we don't have `gsl` is suspicious.

Reply:
It should be Linux-only, I missed that.